src/pretix/presale/style.py

```diff
@@ -225,17 +226,67 @@ def regenerate_organizer_css(organizer_id: int, regenerate_events=True):
 }
 """
 
+register_event_fonts = EventPluginSignal()
+"""
+Return a dictionaries of the following structure. Paths should be relative to static root.
+As with all plugin signals, the ``sender`` keyword argument will contain the event.
+{
+    "font name": {
+        "regular": {
+            "truetype": "….ttf",
+            "woff": "…",
+            "woff2": "…"
+        },
+        "bold": {
+            ...
+        },
+        "italic": {
+            ...
+        },
+        "bolditalic": {
+            ...
+        },
+        "pdf_only": False,   # if True, font is not usable on the web,
+    }
+}
+"""
+
+
+def get_fonts(event: Event = None, pdf_only=False):
+    def nested_dict_values(d):
+        for v in d.values():
+            if isinstance(v, dict):
+                yield from nested_dict_values(v)
+            else:
+                if isinstance(v, str):
+                    yield v
 
-def get_fonts():
     f = {}
+    received_fonts = {}
+
     for recv, value in register_fonts.send(0):
-        f.update(value)
+        received_fonts.update(value)
+
+    if event:
+        for recv, value in register_event_fonts.send(event):
+            received_fonts.update(value)
+
+    for font, payload in received_fonts.items():
+        if pdf_only:
+            if any(v.startswith('https') for v in list(nested_dict_values(payload))):
```

Comment:
see above, I'd either go with ``"//" in`` or ``.startswith('https://')``. Your approach is just calling for a font named "https font" ;)